gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_model.dart

```diff
@@ -35,8 +52,29 @@ class SubscribeNowModel {
     }
   }
 
-  /// Returns true if the environment variable 'UP4W_ALLOW_STORE_PURCHASE' has been set.
-  /// Since this reading won't change during the app lifetime, even if the user changes
-  /// it's value from outside, the value is cached so we don't check the environment more than once.
-  bool get purchaseAllowed => _isPurchaseAllowed;
+  void tokenUpdate(String token) async {
+    _token.update(token);
+    notifyListeners();
+  }
+}
+
+/// A [ProToken] with validation.
+///
+/// Similar to a [EitherValueNotifier], without the [ValueNotifier].
+class ProTokenValue {
+  ProTokenValue() : either = const Either.left(TokenError.empty);
+
+  Either<TokenError, ProToken?> either;
+  ProToken? get token => either.orNull();
+  String? get value => either.orNull()?.value;
+  TokenError? get error => either.fold(ifLeft: (e) => e, ifRight: (_) => null);
+  bool get hasError => either.isLeft;
+
+  void update(String token) {
+    either = ProToken.create(token);
+  }
+
+  void clear() {
+    either = const Either.right(null);
+  }
 }
```

Comment:
We can get rid of this class and use the Either directly.

Reply:
I suspect we can also delete the EitherValueNotifier, but I'm not 100% sure. You should check if that's possible and do accordingly.